File: Wiggle Sort2/solution.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
	public:
		void wiggleSort(vector<int>& nums) {
			if(nums.size()==0)
				return;
			sort(nums.begin(),nums.end());
			int point;
			if(nums.size()%2)
				point=nums.size()/2;
			else
				point=nums.size()/2-1;
			vector<int> small(nums.begin(),nums.begin()+point+1);
			vector<int> large(nums.begin()+point+1,nums.end());
			for(int i=0;i<nums.size();i++){
				if(i%2==0){
					nums[i]=small[small.size()-i/2-1];
				}else{
					nums[i]=large[large.size()-i/2-1];
				}
			}
		}
};
int main(){
	vector<int> nums={4,5,5,6};
	Solution *sol=new Solution;
	sol->wiggleSort(nums);
	for(int num:nums)
		cout<<num<<endl;
}

File: Wiggle Sort2/Solution.java
import java.util.*;
public class Solution {
	public void wiggleSort(int[] nums) {
		Arrays.sort(nums);
		int[] tmp= new int[nums.length];
		int low=(nums.length+1)/2;
		int high=nums.length;
		for(int i=0;i<nums.length;i++){
			if(i%2==0){
				tmp[i]=nums[--low];
			} else{
				tmp[i]=nums[--high];
			}

		}		
		for(int i=0;i<nums.length;i++)
			nums[i]=tmp[i];
	}
	public static void main(String[] args){
		Solution sol=new Solution();
		int[] nums={4,4,5,6};
		sol.wiggleSort(nums);
		for(int num:nums)
			System.out.print(num);
		System.out.println("");
	}
}
